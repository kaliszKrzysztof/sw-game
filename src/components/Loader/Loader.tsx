import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { makeStyles, createStyles, fade } from '@material-ui/core/styles';

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      position: 'absolute',
      bottom: 0,
      top: 0,
      left: 0,
      right: 0,
      backgroundColor: fade('#000', 0.3),
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      zIndex: 1,
    },
  }),
);

interface LoaderProps {
  size?: number;
}

const Loader: React.FC<LoaderProps> = ({ size = 40 }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <CircularProgress size={size} data-testid="loader" />
    </div>
  );
};

export default Loader;
