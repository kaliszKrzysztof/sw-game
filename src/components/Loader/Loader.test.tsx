import React from 'react';
import { render } from '@testing-library/react';
import Loader from './Loader';

describe('<Loader />', () => {
  it('should render', () => {
    render(<Loader />);
  });
  it('matches the snapshot without props', () => {
    const { container, getByTestId } = render(<Loader />);
    expect(getByTestId('loader')).toHaveAttribute(
      'style',
      'width: 40px; height: 40px;',
    );
    expect(container).toMatchSnapshot();
  });

  it('passes props to material-ui loader', () => {
    const { container, getByTestId } = render(<Loader size={100} />);
    expect(getByTestId('loader')).toHaveAttribute(
      'style',
      'width: 100px; height: 100px;',
    );
    expect(container).toMatchSnapshot();
  });
});
