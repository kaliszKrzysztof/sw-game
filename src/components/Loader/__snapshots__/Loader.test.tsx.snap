// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> matches the snapshot without props 1`] = `
<div>
  <div
    class="makeStyles-root-2"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      data-testid="loader"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`<Loader /> passes props to material-ui loader 1`] = `
<div>
  <div
    class="makeStyles-root-3"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      data-testid="loader"
      role="progressbar"
      style="width: 100px; height: 100px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
